datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -742,19 +756,15 @@ private Future<Boolean> dispatchAssignmentChangeIfNeeded(String connectorType, L
     ConnectorInfo connectorInfo = _connectors.get(connectorType);
     ConnectorWrapper connector = connectorInfo.getConnector();
 
-    List<DatastreamTask> addedTasks = new ArrayList<>(assignment);
-    List<DatastreamTask> removedTasks;
     List<DatastreamTask> oldAssignment = _assignedDatastreamTasks.values()
         .stream()
         .filter(t -> t.getConnectorType().equals(connectorType))
         .collect(Collectors.toList());
+    List<DatastreamTask> addedTasks = getAddedTasks(assignment, oldAssignment);
+    List<DatastreamTask> removedTasks = getRemovedTasks(assignment, oldAssignment);
 
     // if there are any difference in the list of assignment. Note that if there are no difference
     // between the two lists, then the connector onAssignmentChange() is not called.
```

Comment:
The refactored code got moved before the comment, is that intentional?

Reply:
Lines 745, 746, 754 through 757 were removed because I introduced methods that get added/removed tasks. This is done intentionally.